tuf/api/metadata.py

```diff
@@ -1123,6 +1124,41 @@ def to_dict(self) -> Dict[str, Any]:
             **self.unrecognized_fields,
         }
 
+    @classmethod
+    def from_file(cls, filepath: str) -> "TargetFile":
+        """Creates TargetFile object from a file."""
+        # Ensure the file path is valid/exists.
+        if not os.path.exists(filepath):
+            raise exceptions.NotFoundError(repr(filepath) + " does not exist.")
```

Comment:
We should use existing system exceptions when possible (old code uses a lot of custom exceptions, but I don't want this to happen in Metadata API if we can avoid it). Luckily open() already raises FileNotFoundError in this case so we don't need to have this code (but we may want to document the FileNotFoundError).